### PR TITLE
Monkeytyping core libaries.

### PR DESCRIPTION
**Pull request by ankenyr**
This PR monkey types a set of our core libraries 
aerleon/lib/aclgenerator.py
 aerleon/lib/plugin_supervisor.py
aerleon/lib/policy.py
aerleon/lib/policy_simple.py
aerleon/lib/recognizers.py
aerleon/lib/summarizer.py
aerleon/lib/yaml.py

Some functions were not hit due to not being covered by tests. A follow-up PR should either create more tests or create typing manually in the future.

### CODE REVIEW

File: aerleon/lib/policy.py [hunk 1]
```diff
@@ -141,7 +145,9 @@ class InvalidNumericProtoValue(Error):
     """Error when protocols are numeric and not between -1 and 255."""
 
 
-def TranslatePorts(ports, protocols, term_name):
+def TranslatePorts(
+    ports: List[str], protocols: List[Union[str]], term_name: str
```
Comment (jtwb) [inline aerleon/lib/policy.py:149]: `Union[str]` is just `str`

File: aerleon/lib/policy.py [hunk 2]
```diff
@@ -2592,7 +2606,7 @@ def p_strings_or_ints(p):
             p[0] = [p[1]]
 
 
-def p_error(p):
+def p_error(p: LexToken):
```
Comment (jtwb) [inline aerleon/lib/policy.py:2609]: Good catch by monkeytype here - the PLY docs consistently refer to the argument to `p_error` as "p" even though it is a LexToken and not a production. Would be more correct as "t".

File: aerleon/lib/policy.py [hunk 3]
```diff
@@ -2718,8 +2732,13 @@ def ParseFile(filename, definitions=None, optimize=True, base_dir='', shade_chec
 
 
 def ParsePolicy(
-    data, definitions=None, optimize=True, base_dir='', shade_check=False, filename=''
-):
+    data: str,
+    definitions: Optional[MagicMock] = None,
```
Comment (jtwb) [inline aerleon/lib/policy.py:2736]: This is not correct, MagicMock should not be part of the public interface.

File: aerleon/lib/recognizers.py [hunk 4]
```diff
@@ -345,7 +345,7 @@ class TSection(TComposition):
 
     of: "list[typing.Tuple[str | TValue | TUnion, TValue | TComposition]]"
 
-    def Recognize(self, value: dict):
+    def Recognize(self, value: dict) -> typing.Dict[str, typing.List[str]]:
```
Comment (jtwb) [inline aerleon/lib/recognizers.py:348]: We should probably just import Dict, List and Any at this point.

File: aerleon/lib/yaml.py [hunk 5]
```diff
@@ -145,7 +146,9 @@ def ParsePolicy(
     return policy.FromBuilder(PolicyBuilder(policy_dict, definitions, optimize, shade_check))
 
 
-def PreprocessYAMLPolicy(filename, base_dir, policy_dict: PolicyDict):
+def PreprocessYAMLPolicy(
+    filename: str, base_dir: str, policy_dict: PolicyDict
+) -> Optional[Dict[str, List[Dict[str, Union[Dict[str, Dict[str, str]], List[Dict[str, str]]]]]]]:
```
Comment (jtwb) [inline aerleon/lib/yaml.py:151]: This should just be `Optional[PolicyDict]`

**Review by jtwb — APPROVED** [review]
Apart from comments this PR looks very good.

File: aerleon/lib/policy.py (hunk 1 above)
Comment (ankenyr) [reply aerleon/lib/policy.py:149]: Done

File: aerleon/lib/policy.py (hunk 3 above)
Comment (ankenyr) [reply aerleon/lib/policy.py:2736]: Done

File: aerleon/lib/yaml.py (hunk 5 above)
Comment (ankenyr) [reply aerleon/lib/yaml.py:151]: Done

**Pull request merged**